### PR TITLE
Nfdiv 1490 about the other party

### PR DESCRIPTION
**Pull request by mark-butler-solirius**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1490

### Change description ###

Dynamic content changes for "About the other party"

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant2.java [hunk 1]
```diff
@@ -15,23 +15,37 @@ public void addTo(final PageBuilder pageBuilder) {
...
-                .mandatoryWithLabel(Applicant::getGender,
-                        "What is ${labelContentTheApplicant2}'s gender?")
+                .optionalWithLabel(Applicant::getGender,
+                        "Is ${labelContentTheApplicant2} male or female?")
+                .mandatory(
+                    Applicant::getDivorceWho,
+                    "divorceOrDissolution=\"divorce\"",
+                    null,
+                    "What is ${labelContentTheApplicant2}"
+                )
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant2.java:48]: Does this need a `?`

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant2.java:48]: on line 47, sorry

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Applicant.java [hunk 2]
```diff
@@ -79,9 +79,7 @@
     private YesOrNo lastNameChangedWhenMarried;
 
     @CCD(
-        label = "Have they changed their name since they got married?",
-        hint = "Is their current name different to their married name or the name shown on their "
-            + "marriage certificate?"
+        label = "Have they changed their name since they got married?"
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/Applicant.java:82]: for civil partnership married won't be applicable ? could you plz check with Michael on this ?

**Review by tomxelliott — APPROVED** [review]
Looks good, just had one comment

**Review by nitinprabhu — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant2.java (hunk 1 above)
Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant2.java:48]: Yes it does - thanks Tom

Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolAboutApplicant2.java:48]: Pushed fix

**Pull request merged**